ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java

```diff
@@ -30,223 +33,230 @@
-        this.options.addOption(sizeOfFiles);
-        this.options.addOption(bufferSize);
-        this.options.addOption(numberOfThreads);
-        this.options.addOption(ignoreErrors);
-        this.options.addOption(noFollowSymlinks);
-        this.options.addOption(followSymLinks);
-        this.options.addOption(metadata);
-        this.options.addOption(modifyParams);
-        this.options.addOption(discard);
-        this.options.addOption(ignoreNamingConflicts);
-        this.options.addOption(inCache);
-        this.options.addOption(verifyPercent);
+        // parse all values required to select help, query version, or specify COMMAND, set up the client, and configure logging.
+        addOption(ENDPOINT, "endpoint");
+        addOption(ACCESS_KEY, "accessKeyId");
+        addOption(SECRET_KEY, "secret_key");
+        addOption(COMMAND, "command");
+        addOption(PROXY, "proxy");
+        addOption(HTTP, "http");
+        addOption(INSECURE, "insecure");
+        addOption(PRINT_HELP, "help");
+        addOption(COMMAND_HELP, "command_help");
+        addOption(VERBOSE, "verbose");
+        addOption(DEBUG, "debug");
+        addOption(TRACE, "trace");
+        addOption(LOG_VERBOSE, "log-verbose");
+        addOption(LOG_DEBUG, "log-debug");
+        addOption(LOG_TRACE, "log-trace");
+        addOption(PRINT_VERSION, "version");
+        addOption(VIEW_TYPE, "view_type");
+        addOption(RETRIES, "retries");
+        addOption(BUFFER_SIZE, "buffer_size");
+
         this.processCommandLine();
     }
 
-    private void processCommandLine() throws ParseException, BadArgumentException {
+    private String[] rootArgumentsOnly() throws BadArgumentException {
+        // Present only COMMAND, help and version arguments to parser, After the COMMAND
+        // is instantiated, the full parse will be run against the complete Options list.
+        List<String> rootArguments = new ArrayList<String>();
+
+        for (int i = 0; i < this.args.length; i++) {
+            final String token = this.args[i];
+            // allow get COMMAND (-c) and subsequent argument
+            if (token.equals("-c")) {
+                rootArguments.add(token);
+                if (i < this.args.length) {
+                    rootArguments.add(this.args[++i]);
+                } else {
+                    throw new BadArgumentException("No argument supplied for option: " + token);
+                }
+            }
+            // allow get COMMAND help and subsequent argument (unless it is followed by an option (- or --)
+            if (token.equals("--help")) {
+                rootArguments.add(token);
+                // might or might not have arg
+                if ((i < this.args.length) && (!this.args[i + 1].startsWith("-"))) {
+                    rootArguments.add(this.args[++i]);
+                }
+            }
+            // allow version in root parse
+            if ((token.equals("--version")) || (token.equals("-h"))
```

Comment:
Rather than hard coding this is there a way we can build up the set from the default argument list?